core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -96,45 +114,76 @@ oppia.controller('FeedbackTab', [
       size: 'lg',
       resolve: {
-        suggestion: function() {
-          return $scope.activeThread.suggestion;
+        suggestionInfo: function() {
+          return {
+            isSuggestionValid: $scope.isSuggestionValid(),
+            hasUnsavedChanges: $scope.hasUnsavedChanges()
+          };
+        },
+        stateContent: function() {
+          var states = explorationData.data.states;
+          var stateName = $scope.activeThread.suggestion.state_name;
+          return {
+            oldContent: states[stateName].content[0].value,
+            newContent: $scope.activeThread.suggestion.state_content.value
+          };
         }
       },
       controller: [
-        '$scope', '$modalInstance', 'suggestion',
-        function($scope, $modalInstance, suggestion) {
-        $scope.oldContent = suggestion.state_content.old_content;
-        $scope.newContent = suggestion.state_content.new_content;
+        '$scope', '$modalInstance', 'suggestionInfo', 'stateContent',
+        function($scope, $modalInstance, suggestionInfo, stateContent) {
+        var ACTION_INVALID_MSG = 'This suggestion has already been ' +
+          'acted upon, or was made for an outdated version '  +
+          'of this exploration.';
+        var UNSAVED_CHANGES_MSG = 'You have unsaved changes to this ' +
+          'exploration. Please save your changes before acting on ' +
+          'a suggestion.';
+        $scope.canActOnSuggestion = (
+          suggestionInfo.isSuggestionValid && !suggestionInfo.hasUnsavedChanges
+        );
+        $scope.errorMessage = suggestionInfo.isSuggestionValid
+          ? UNSAVED_CHANGES_MSG : ACTION_INVALID_MSG;
+        $scope.oldContent = stateContent.oldContent;
+        $scope.newContent = stateContent.newContent;
+        $scope.commitMessage = '';
 
         $scope.acceptSuggestion = function() {
           $modalInstance.close({
-            action: 'accept'
+            action: ACTION_ACCEPT_SUGGESTION,
+            commitMsg: $scope.commitMessage
           });
         };
 
         $scope.rejectSuggestion = function() {
           $modalInstance.close({
-            action: 'reject'
+            action: ACTION_REJECT_SUGGESTION
           });
         };
 
         $scope.cancelReview = function() {
-          $modalInstance.dismiss('cancel');
+          $modalInstance.dismiss();
         };
       }]
     }).result.then(function(result) {
       threadDataService.resolveSuggestion(
-        $scope.activeThread.suggestion.suggestion.id, result.action);
-      // Update the status of the feedback thread.
-      var msg, status;
-      if (result.action === 'accept') {
-        msg = 'Suggestion accepted.';
-        status = 'fixed';
-      } else {
-        msg = 'Suggestion rejected.';
-        status = 'not_actionable';
-      }
-      $scope.addNewMessage($scope.activeThread.thread_id, msg, status);
+        $scope.activeThread.thread_id, result.action, result.commitMsg,
+        function(res) {
```

Comment:
Ah, I wasn't aware about the linter. I've fixed this (and other linter issues raised in this file).